src/test/java/com/stripe/model/ChargeTest.java

```diff
@@ -0,0 +1,151 @@
+        assertEquals(expandedCharge.getDestination(), "newidhere");
+        Account newDestination = new Account();
+        expandedCharge.setDestinationObject(newDestination);
+        assertEquals(expandedCharge.getDestinationObject(), newDestination);
+    }
+
+    @Test
+    public void testDeserializeExpandInvoice() throws IOException {
+        Invoice invoice = expandedCharge.getInvoiceObject();
+        assertEquals(expandedCharge.getInvoice(), "in_199Llh2eZvKYlo2CMA8qc7Ng");
+        assertEquals(invoice.getId(), "in_199Llh2eZvKYlo2CMA8qc7Ng");
+        assertEquals((long) invoice.getAmountDue(), 999);
+        //Setters:
+        expandedCharge.setInvoice("newidhere");
+        assertEquals(expandedCharge.getInvoice(), "newidhere");
+        Invoice newInvoice = new Invoice();
+        expandedCharge.setInvoiceObject(newInvoice);
+        assertEquals(expandedCharge.getInvoiceObject(), newInvoice);
+    }
+
+    @Test
+    public void testDeserializeExpandOrder() throws IOException {
+        Order order = expandedCharge.getOrderObject();
+        assertEquals(expandedCharge.getOrder(), "or_199IRC2eZvKYlo2CPfFd4CB6");
+        assertEquals(order.getId(), "or_199IRC2eZvKYlo2CPfFd4CB6");
+        assertEquals((long) order.getAmount(), 1500);
+        //Setters:
+        expandedCharge.setOrder("newidhere");
+        assertEquals(expandedCharge.getOrder(), "newidhere");
+        Order newOrder = new Order();
+        expandedCharge.setOrderObject(newOrder);
+        assertEquals(expandedCharge.getOrderObject(), newOrder);
+    }
+
+    @Test
+    public void testDeserializeExpandSourceTransfer() throws IOException {
+        Transfer sourceTransfer = expandedCharge.getSourceTransferObject();
+        assertEquals(expandedCharge.getSourceTransfer(), "tr_199IPs2eZvKYlo2C9hMneg1R");
+        assertEquals(sourceTransfer.getId(), "tr_199IPs2eZvKYlo2C9hMneg1R");
+        assertEquals((long) sourceTransfer.getAmount(), 400);
+        //Setters:
+        expandedCharge.setSourceTransfer("newidhere");
+        assertEquals(expandedCharge.getSourceTransfer(), "newidhere");
+        Transfer newST = new Transfer();
+        expandedCharge.setSourceTransferObject(newST);
+        assertEquals(expandedCharge.getSourceTransferObject(), newST);
+    }
+
+    @Test
+    public void testDeserializeExpandTransfer() throws IOException {
+        Transfer transfer = expandedCharge.getTransferObject();
+        assertEquals(expandedCharge.getTransfer(), "tr_199IPs2eZvKYlo2C9hMneg1R");
+        assertEquals(transfer.getId(), "tr_199IPs2eZvKYlo2C9hMneg1R");
+        assertEquals((long) transfer.getAmount(), 400);
+        //Setters:
+        expandedCharge.setTransfer("newidhere");
+        assertEquals(expandedCharge.getTransfer(), "newidhere");
+        Transfer newTransfer = new Transfer();
+        expandedCharge.setTransferObject(newTransfer);
+        assertEquals(expandedCharge.getTransferObject(), newTransfer);
+    }
+}
```

Comment:
Thanks for writing all these tests!

I suspect that thoroughly vetting just the core expansion code is going to be mostly adequate for our purposes. Adding a test for every possible expansion is really just testing the API rather than our own code and I'm not totally convinced that's necessary. Given that they're already written, I'd leave them, but I don't think we should require a new test case for every possible expansion for future patches.